src/pages/settings/email/index.vue

```diff
@@ -77,10 +80,23 @@ export default {
     hasEmail() {
       return !!this.walletEmail || !!this.walletEmailUnverified;
     },
+    isClaimingPendingNftMode() {
+      return !!this.$route.query.claim_pending_nft && !this.hasEmail;
+    },
     isDisabledChangingEmail() {
       return !this.newEmail || this.newEmail === this.walletEmailUnverified;
     },
   },
+  watch: {
+    isClaimingPendingNftMode: {
+      immediate: true,
+      handler(newVal) {
+        if (newVal && this.$route.query.email) {
+          this.newEmail = this.$route.query.email;
```

Comment:
`this.newEmail = this.$route.query.email;` we can always assign this in data regard less `isClaimingPendingNftMode` or not